src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -243,6 +244,11 @@ public static int PopCount(uint value)
                 return (int)Popcnt.PopCount(value);
             }
 
+            if (AdvSimd.IsSupported)
+            {
+                return PopCount((ulong)value);
```

Comment:
Can you share the code that jit generates here?
Does it inline PopCount(ulong) here?

Would it be better to have the following instead and add a comment explaining why cast to ulong is needed
```c#
Vector64<ulong> input = Vector64.Create((ulong)value);
Vector64<byte> aggregated = AdvSimd.Arm64.AddAcross(AdvSimd.PopCount(input.AsByte()));
return AdvSimd.Extract(aggregated, 0);
```

Reply:
I think I can add a comment about casting to `ulong` but still call the `PopCount(ulong)` version so we have code at single place.

Reply:
I will let you to decide what approach to take here.

IMO, it looks slightly weird that we check for AdvSimd.Arm64.IsSupported before calling to PopCount(ulong) - basically, assuming PopCount(ulong) implementation.

Reply:
Now I think about it, it makes sense to have dedicated implementation. I am not sure if this will ever happen, but if `AdvSimd.IsSupported == true` but `AdvSimd.Arm64.IsSupported == false`, we would end up doing software fallback implemented for long which has larger constants to load and might end up more instructions with the movz/movk. So I think I will just do it the way you have suggested above.

Reply:
> Can you share the code that jit generates here?

Here you go.
<details>
<summary> Generated code for uint and ulong </summary>

```asm
; Assembly listing for method projs.PopCountTester:PPTest_uint(int):int
; Emitting BLENDED_CODE for generic ARM64 CPU - Windows
; optimized code
; fp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  3,  3   )     int  ->   x0
;# V01 OutArgs      [V01    ] (  1,  1   )  lclBlk ( 0) [sp+0x00]   "OutgoingArgSpace"
;  V02 tmp1         [V02,T01] (  2,  2   )     int  ->   x0         "Inline return value spill temp"
;  V03 tmp2         [V03,T02] (  2,  2   )   simd8  ->  d16         HFA(double)  "Inline stloc first use temp"
;  V04 tmp3         [V04,T03] (  2,  2   )   simd8  ->  d16         HFA(double)  "Inline stloc first use temp"
;
; Lcl frame size = 0

G_M7171_IG01:
        A9BF7BFD          stp     fp, lr, [sp,#-16]!
        910003FD          mov     fp, sp
                                                ;; bbWeight=1    PerfScore 1.50
G_M7171_IG02:
        2A0003E0          mov     w0, w0
        4E081C10          mov     v16.d[0], x0
        0E205A10          cnt     v16.8b, v16.8b
        0E31BA10          addv    b16, v16.8b
        0E013E00          umov    w0, v16.b[0]
                                                ;; bbWeight=1    PerfScore 5.50
G_M7171_IG03:
        A8C17BFD          ldp     fp, lr, [sp],#16
        D65F03C0          ret     lr
                                                ;; bbWeight=1    PerfScore 2.00

; Total bytes of code 36, prolog size 8, PerfScore 12.60, (MethodHash=907ee3fc) for method projs.PopCountTester:PPTest_uint(int):int
; ============================================================

; Assembly listing for method projs.PopCountTester:PPTest_ulong(long):int
; Emitting BLENDED_CODE for generic ARM64 CPU - Windows
; optimized code
; fp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  3,  3   )    long  ->   x0
;# V01 OutArgs      [V01    ] (  1,  1   )  lclBlk ( 0) [sp+0x00]   "OutgoingArgSpace"
;  V02 tmp1         [V02,T01] (  2,  2   )     int  ->   x0         "Inline return value spill temp"
;  V03 tmp2         [V03,T02] (  2,  2   )   simd8  ->  d16         HFA(double)  "Inline stloc first use temp"
;  V04 tmp3         [V04,T03] (  2,  2   )   simd8  ->  d16         HFA(double)  "Inline stloc first use temp"
;
; Lcl frame size = 0

G_M14403_IG01:
        A9BF7BFD          stp     fp, lr, [sp,#-16]!
        910003FD          mov     fp, sp
                                                ;; bbWeight=1    PerfScore 1.50
G_M14403_IG02:
        4E081C10          mov     v16.d[0], x0
        0E205A10          cnt     v16.8b, v16.8b
        0E31BA10          addv    b16, v16.8b
        0E013E00          umov    w0, v16.b[0]
                                                ;; bbWeight=1    PerfScore 5.00
G_M14403_IG03:
        A8C17BFD          ldp     fp, lr, [sp],#16
        D65F03C0          ret     lr
                                                ;; bbWeight=1    PerfScore 2.00

; Total bytes of code 32, prolog size 8, PerfScore 11.70, (MethodHash=3f67c7bc) for method projs.PopCountTester:PPTest_ulong(long):int
; ============================================================

```
</details>